src/apps/btc/btc_common.c

```diff
@@ -356,3 +356,49 @@ bool btc_common_pkscript_from_outputhash(
     }
     return true;
 }
+
+bool btc_common_pkscript_from_multisig(
+    const BTCScriptConfig_Multisig* multisig,
+    uint32_t keypath_change,
+    uint32_t keypath_address,
+    uint8_t* script_out,
+    size_t* script_out_size)
+{
+    uint8_t pubkeys[sizeof(multisig->xpubs) / sizeof(XPub) * EC_PUBLIC_KEY_LEN];
+
+    for (size_t index = 0; index < multisig->xpubs_count; index++) {
+        const XPub* xpub_in = &multisig->xpubs[index];
+        struct ext_key xpub = {0};
+        if (!apps_common_bip32_xpub_from_protobuf(xpub_in, &xpub)) {
+            return false;
+        }
+        struct ext_key derived_cosigner_xpub = {0};
+        const uint32_t keypath[2] = {keypath_change, keypath_address};
+        if (bip32_key_from_parent_path(
+                &xpub, keypath, 2, BIP32_FLAG_KEY_PUBLIC, &derived_cosigner_xpub) != WALLY_OK) {
+            return false;
+        }
+        memcpy(
+            &pubkeys[index * EC_PUBLIC_KEY_LEN], derived_cosigner_xpub.pub_key, EC_PUBLIC_KEY_LEN);
+    }
+
+    size_t written;
+    if (wally_scriptpubkey_multisig_from_bytes(
```

Comment:
What happens if we receive an input with `xpubs_count == 0`? Does libwally behave correctly with an input of size 0? (Also see below)

Reply:
Yes it does, although the docstring still says that the multisig params must be validated upfront (another function to do that is coming soon).